cmd/trace-agent/windows/controlsvc/controlsvc.go

```diff
@@ -0,0 +1,88 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build windows
+// +build windows
+
+// Package controlsvc contains shared code for controlling the Windows agent service.
+package controlsvc
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/config"
+
+	"golang.org/x/sys/windows/svc"
+	"golang.org/x/sys/windows/svc/mgr"
+)
+
+// StartService starts the agent service via the Service Control Manager
+func StartService() error {
+	m, err := mgr.Connect()
+	if err != nil {
+		return err
+	}
+	defer m.Disconnect()
+	s, err := m.OpenService(config.ServiceName)
+	if err != nil {
+		return fmt.Errorf("could not access service: %v", err)
+	}
+	defer s.Close()
+	err = s.Start("is", "manual-started")
+	if err != nil {
+		return fmt.Errorf("could not start service: %v", err)
+	}
+	return nil
+}
+
+// RestartService restarts the agent service by calling StopService and StartService
+func RestartService() error {
+
+	var err error
+	if err = StopService(); err == nil {
+		err = StartService()
+	}
+
+	return err
+}
+
+// StopService stops the agent service via the Service Control Manager
+func StopService() error {
+	return controlService(svc.Stop, svc.Stopped)
+}
+
+// TODO: this should be migrated to use winutil.controlsvc
+func controlService(c svc.Cmd, to svc.State) error {
```

Comment:
Something for this PR, or a follow-up PR? (i.e. is this a release-blocking TODO?)

Reply:
This is not for this PR; definitely not.